chia/full_node/full_node_api.py

```diff
@@ -1650,5 +1642,157 @@ def get_fee_estimates(est: FeeEstimatorInterface, req_times: List[uint64]) -> Li
+        added = subs.add_coin_subscriptions(peer_id, list(new_subscriptions), max_subscriptions)
+
+        response = wallet_protocol.RespondAddCoinSubscriptions(list(added))
+        msg = make_msg(ProtocolMessageTypes.respond_add_coin_subscriptions, response)
+        return msg
+
+    @api_request(
+        peer_required=True,
+        reply_types=[ProtocolMessageTypes.respond_remove_ph_subscriptions],
+    )
+    async def request_remove_ph_subscriptions(
+        self, request: wallet_protocol.RequestRemovePhSubscriptions, peer: WSChiaConnection
+    ) -> Message:
+        peer_id = peer.peer_node_id
+        subs = self.full_node.subscriptions
+
+        removed = subs.remove_ph_subscriptions(peer_id, request.puzzle_hashes)
+
+        response = wallet_protocol.RespondRemovePhSubscriptions(list(removed))
+        msg = make_msg(ProtocolMessageTypes.respond_remove_ph_subscriptions, response)
+        return msg
+
+    @api_request(
+        peer_required=True,
+        reply_types=[ProtocolMessageTypes.respond_remove_coin_subscriptions],
+    )
+    async def request_remove_coin_subscriptions(
+        self, request: wallet_protocol.RequestAddCoinSubscriptions, peer: WSChiaConnection
+    ) -> Message:
+        peer_id = peer.peer_node_id
+        subs = self.full_node.subscriptions
+
+        removed = subs.remove_coin_subscriptions(peer_id, request.coin_ids)
+
+        response = wallet_protocol.RespondRemoveCoinSubscriptions(list(removed))
+        msg = make_msg(ProtocolMessageTypes.respond_remove_coin_subscriptions, response)
+        return msg
+
+    @api_request(
+        peer_required=True,
+        reply_types=[ProtocolMessageTypes.respond_reset_subscriptions],
+    )
+    async def request_reset_subscriptions(
+        self, _request: wallet_protocol.RequestResetSubscriptions, peer: WSChiaConnection
+    ) -> Message:
+        peer_id = peer.peer_node_id
+        subs = self.full_node.subscriptions
+
+        ph_subs = subs.puzzle_subscriptions(peer_id)
+        coin_subs = subs.coin_subscriptions(peer_id)
+        subs.remove_peer(peer_id)
+
+        response = wallet_protocol.RespondResetSubscriptions(list(ph_subs), list(coin_subs))
+        msg = make_msg(ProtocolMessageTypes.respond_reset_subscriptions, response)
+        return msg
+
+    @api_request(peer_required=True)
+    async def request_ph_state(self, request: wallet_protocol.RequestPhState, peer: WSChiaConnection) -> None:
+        peak = self.full_node.blockchain.get_peak()
+        assert peak is not None
+
+        try:
+            async with self.full_node.stream_coin_state_sem.acquire():
+                stream = self.full_node.coin_store.stream_coin_states_by_puzzle_hashes(
+                    set(request.puzzle_hashes),
+                    request.min_height,
+                    request.filters.include_spent,
+                    request.filters.include_unspent,
+                    request.filters.include_hinted,
```

Comment:
this won't work if you "blindly" accept my change above, so you should probably make the change yourself and make everything work before updating the PR